framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileParser.java

```diff
@@ -1816,6 +1815,18 @@ private void annotateTypeParameters(
                     stubWarnNotFound(
                             param, "Annotations on intersection types are not yet supported");
                 }
+                if (param.getTypeBound().size() == 1
+                        && param.getTypeBound().get(0).getAnnotations().isEmpty()
+                        && paramType
+                                .getUpperBound()
+                                .getUnderlyingType()
+                                .toString()
+                                .contentEquals("java.lang.Object")) {
+                    // If there is an explicit "T extends Object" type parameter bound,
+                    // treat it like an explicit use of "Object" in code.
+                    AnnotatedTypeMirror ub = atypeFactory.getAnnotatedType(Object.class);
+                    paramType.getUpperBound().addAnnotations(ub.getAnnotations());
```

Comment:
Could we need to first remove any existing annotations? Granted, it's far less likely that such annotations would exist, particularly in the JDK :) But in principle, stubs _could_ be used to override a source `<T extends @Nullable Object>` with plain `<T extends Object>`.

Reply:
In https://github.com/eisop/checker-framework/pull/247/commits/c8239840b7f28fdcf50cc934f098d86e56d1ac99 I've added a test where the bytecode contains an explicit Nullable bound and the same four combinations of stubfiles. The tests work correctly, but I still changed the `addAnnotations` to `replaceAnnotations`, for consistency with the rest of the file.

Reply:
Thanks for checking! I can believe it works; I didn't try to dig deeply enough to see where the original objects came from and whether they might have already been annotated.